include/hw_unique_key.h

```diff
@@ -20,15 +20,62 @@
 extern "C" {
 #endif
 
+#include <device.h>
+
+#if defined(CONFIG_HAS_HW_NRF_CC310)
+#define HUK_SIZE_WORDS 4
+#elif defined(CONFIG_HAS_HW_NRF_CC312)
+#define HUK_SIZE_WORDS 8
+#endif
+#define HUK_SIZE_BYTES (HUK_SIZE_WORDS * 4)
+
+/* keyslots are 1-indexed so this is the first. */
+#if defined(CONFIG_HAS_HW_NRF_CC310)
+#define HUK_KEYSLOT_KDR  1
+#endif
+#define HUK_KEYSLOT_MKEK 3
+#define HUK_KEYSLOT_MEXT 5
+
+static const uint32_t slots[] = {
+#ifdef CONFIG_HAS_HW_NRF_CC310
+	HUK_KEYSLOT_KDR,
+#endif
+#if defined(CONFIG_SOC_NRF9160) || defined(CONFIG_SOC_NRF5340_CPUAPP)
+	HUK_KEYSLOT_MKEK,
+	HUK_KEYSLOT_MEXT,
+#endif
+};
+
+#define UICR_KEY_INDEX(KEYSLOT) ((KEYSLOT) - 1) /* UICR KEYSLOT indices are 0-indexed. */
+
+/**
+ * @brief Write a Hardware Unique Key to the KMU. This can only be done once
+ *       (until a mass erase happens). Panic on failure.
+ *
+ * @param[in] key  The key to write. Must be @ref HUK_SIZE_BYTES bytes long.
+ */
+void hw_unique_key_write(uint32_t kmu_keyslot, const uint8_t *key);
+
+/**
+ * @brief Read a random number from @ref mbedtls_hardware_poll and write it
```

Comment:
mbedtls_hardware_poll gives TRNG. We need to use a CSPRNG

Reply:
Fixed now.